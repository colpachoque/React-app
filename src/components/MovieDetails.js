import React from 'react';
import styles from './MovieDetails.css';
import { Button } from '@material-ui/core';

class MovieDetails extends React.Component {
    render () {
        const { movie } = this.props;
        return (
            <div className={styles.movieCard}>
                <div className={styles.movieImg}></div>
                <div className={styles.movieInfo}>
                    <div className={styles.name}>{movie.name}</div>
                    <div style={{display: 'flex'}}>
                        <div className={styles.year}>{movie.year}</div>
                        <div className={styles.genre}>{movie.duration} min</div>
                    </div>
                    <div className={styles.description}>{movie.description}</div>
                </div>
                <div>
                    <Button>Search</Button>
                </div>
            </div>
        );
    }
}

export default MovieDetails;