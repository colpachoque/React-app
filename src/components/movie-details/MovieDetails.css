.movieCard {
    width: 40%;
    margin: 0 auto;
    display: flex;
}

.movieImg {
    min-width: 250px;
    height: 320px;
    background-image: url('../../../resources/killBill.png');
    background-repeat: no-repeat;
    background-size: cover;
    margin-bottom: 10px;
}

.movieInfo {
    display: flex;
    flex-direction: column;
    margin: 20px;
}

.name {
    font-weight: 600;
    font-size: 40px;
    text-transform: uppercase;
    width: 50%;
}

.duration {
    width: 50%;
}

.year {
    width: 50%;
}

.description {
    font-size: 15px;
    margin-top: 15px;
}
