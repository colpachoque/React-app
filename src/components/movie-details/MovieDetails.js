import React from "react";
import styles from "./MovieDetails.css";
import { Button } from "@material-ui/core";
import { connect } from "react-redux";
import { getMovie } from "../../actions/movies.action";
import { HashRouter as Router, Route, NavLink, Switch } from "react-router-dom";
import NotFound from "../not-found/NotFound";
class MovieDetails extends React.Component {
  componentDidUpdate(prevProps) {
    if (prevProps.location.pathname !== this.props.location.pathname) {
      this.props.getMovie(this.props.match.params.id);
    }
  }

  componentWillMount() {
    this.props.getMovie(this.props.match.params.id);
  }

  render() {
    const { movie } = this.props;
    return (
      <>
        {this.props.movie ? (
          <div className={styles.movieCard}>
            <div className={styles.movieImg} />
            <div className={styles.movieInfo}>
              <div className={styles.name}>{movie.title}</div>
              <div style={{ display: "flex" }}>
                <div className={styles.year}>{movie.release_date}</div>
                <div className={styles.genre}>{movie.runtime} min</div>
              </div>
              <div className={styles.description}>{movie.description}</div>
            </div>
            <div>
              <Button>
                <NavLink to="/">Search</NavLink>
              </Button>
            </div>
          </div>
        ) : (
          <Route exact path="*" component={NotFound} />
        )}
      </>
    );
  }
}

const mapStateToProps = ({ movies: { movie } }) => {
  return { movie };
};
export default connect(
  mapStateToProps,
  { getMovie }
)(MovieDetails);
