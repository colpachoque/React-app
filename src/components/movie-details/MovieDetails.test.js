import React from "react";
import MovieDetails from "./MovieDetails";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

describe("Movie Details component", () => {
  let movie;

  beforeAll(() => {
    movie = {
      name: "The best movie ever",
      genre: "Drama",
      year: 1994,
      description: "The best movie ever"
    };
  });

  it("should be render correctly", () => {
    const component = shallow(<MovieDetails movie={movie} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it("should render correct prop movie", () => {
    const component = mount(<MovieDetails movie={movie} />);
    expect(component.prop("movie")).toEqual(movie);
  });
});
