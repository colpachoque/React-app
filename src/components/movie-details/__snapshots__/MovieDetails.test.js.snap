// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Details component should be render correctly 1`] = `
<div
  className="movieCard"
>
  <div
    className="movieImg"
  />
  <div
    className="movieInfo"
  >
    <div
      className="name"
    >
      The best movie ever
    </div>
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <div
        className="year"
      >
        1994
      </div>
      <div
        className="genre"
      >
         min
      </div>
    </div>
    <div
      className="description"
    >
      The best movie ever
    </div>
  </div>
  <div>
    <WithStyles(Button)>
      Search
    </WithStyles(Button)>
  </div>
</div>
`;
